### PR TITLE
Improve What's Missing AAA article

### PR DESCRIPTION
**Pull request by JohnVeness**
Lots of minor typo/language improvements.

I also think "disk" should be changed to something like "storage" in the Streaming section, to better encompass mobile and cartridge-based systems, but didn't make that change. Maybe you can consider it.

### CODE REVIEW

File: collections/_article/whats-missing-in-godot-for-aaa.md [hunk 1]
```diff
@@ -31,15 +31,15 @@ Several features were added to Godot Physics since 3.x, such as soft bodies and
 
 The custom physics engine still has a considerable amount of issues remaining but we are working hard to ensure it is in a decent state for shipping when 4.0 reaches stability. It will continue seeing improvements afterwards, during the following 4.x release cycles.
 
-That said, Godot 4.0 introduces the ability to bind custom physics engines at runtime (without recompiling Godot) via GDExtension, so it's perfectly possible for the community to integrate other engines such as PhysX, Jolt, or Box2D if need to be.
+That said, Godot 4.0 introduces the ability to bind custom physics engines at runtime (without recompiling Godot) via GDExtension, so it's perfectly possible for the community to integrate other engines such as PhysX, Jolt, or Box2D if needs be.
```
Comment (Calinou) [inline collections/_article/whats-missing-in-godot-for-aaa.md]: The correct form is actually "if need be":

```suggestion
That said, Godot 4.0 introduces the ability to bind custom physics engines at runtime (without recompiling Godot) via GDExtension, so it's perfectly possible for the community to integrate other engines such as PhysX, Jolt, or Box2D if need be.
```

That said, "if needed" is probably easier to understand and is more commonly used.

**Review by Calinou — APPROVED** [review]

**Comment by Calinou** [comment]
The rest of the changes look good to me :slightly_smiling_face:

File: collections/_article/whats-missing-in-godot-for-aaa.md (hunk 1 above)
Comment (JohnVeness) [reply collections/_article/whats-missing-in-godot-for-aaa.md]: Yes, I think "if needed" is best.

**Comment by JohnVeness** [comment]
Thanks :). Any comment on the disk->storage change I mentioned at top, @Calinou ?

**Comment by coppolaemilio** [comment]
Thank you!

**Pull request merged**